### ISSUE TITLE
Debug stepping inconsistencies for functions

### ISSUE THREAD

**Issue by jensjoha** [dart-lang/sdk#29066]
E.g.

```
fooXYZ3() {
  return "fooXYZ";
}

fooXYZ4(int i, int j) {
  return "fooXYZ${i}${j}";
}
```

(i.e. top-level) will stop (when stepping into the function) first at column 8 (in fooXYZ3, i.e. on the open-paren) and 20 (in fooXYZ4, i.e. on the last parameter name).

```
class Bar {
  barXYZ3() {
    return "barXYZ";
  }

  barXYZ4(int i, int j) {
    return "barXYZ${i}${j}";
  }
}
```

(i.e. class functions) will instead stop (when stepping into the function) at column 3 (in barXYZ3, i.e.  on the function name) and 20 (in barXYZ4, i.e. on the last parameter name).

I.e. on functions without parameters the debugger stops either before the name or on the open parenthesis depending on whether it's in a class or not.

**Comment by mhausner** [comment]
https://codereview.chromium.org/2751523009/

**Comment by mhausner** [comment]
Fixed in e78b828d101c24f7e2f0c005a38a2b0555d879a7.
